common/analysis/processingStepCalculation.js

```diff
@@ -1,34 +1,72 @@
+import { uuidv4 } from '@core/uuid'
+import * as R from 'ramda'
+
 import * as ObjectUtils from '@core/objectUtils'
+import * as Validation from '@core/validation/validation'
 
 export const keys = {
   index: ObjectUtils.keys.index,
   uuid: ObjectUtils.keys.uuid,
   nodeDefUuid: ObjectUtils.keys.nodeDefUuid,
   processingStepUuid: 'processingStepUuid',
   props: ObjectUtils.keys.props,
+  temporary: 'temporary', // True when the calculation has been created but not persisted yet
 }
 
 export const keysProps = {
-  formula: 'formula',
+  type: 'type',
   aggregateFn: 'aggregateFn',
+  formula: 'formula',
+  labels: ObjectUtils.keysProps.labels,
 }
 
-const aggregateFn = {
+export const type = {
+  quantitative: 'quantitative',
+  categorical: 'categorical',
+}
+
+export const aggregateFn = {
+  sum: 'sum',
   avg: 'avg',
-  count: 'count',
-  max: 'max',
+  cnt: 'cnt',
   min: 'min',
-  sum: 'sum',
+  max: 'max',
+  med: 'med',
 }
+
+// ====== CREATE
+
+export const newProcessingStepCalculation = (processingStepUuid, index) => ({
```

Comment:
this method is already available in ProcessingChain

Reply:
right, I'll remove it